torch/testing/_internal/common_utils.py

```diff
@@ -1535,11 +1535,20 @@ def random_fullrank_matrix_distinct_singular_value(matrix_size, *batch_dims,
     silent = kwargs.get("silent", False)
     if silent and not torch._C.has_lapack:
         return torch.ones(matrix_size, matrix_size, dtype=dtype, device=device)
+    symmetric = kwargs.get("symmetric", False)
 
     A = torch.randn(batch_dims + (matrix_size, matrix_size), dtype=dtype, device=device)
     u, _, v = A.svd()
-    s = torch.arange(1., matrix_size + 1, dtype=dtype, device=device).mul_(1.0 / (matrix_size + 1)).diag()
-    return u.matmul(s.expand(batch_dims + (matrix_size, matrix_size)).matmul(v.transpose(-2, -1)))
+    if dtype.is_complex:
+        real_dtype = torch.float32 if dtype is torch.complex64 else torch.float64
+    else:
+        real_dtype = dtype
+    s = ((1. - matrix_size) * torch.rand(matrix_size, dtype=real_dtype, device=device) + matrix_size).diag()
```

Comment:
can you explain mathematically what's going on here and why it's the correct thing to do?

Reply:
The random singular values (samples from a uniform distribution with values between 1.0 and `matrix_size`) being used for constructing the random matrix. Previously it was deterministic, so it just introduces more randomness to the test suit. Looking at it now, I see that this change is maybe unnecessary and not related to this PR. I am going to revert this change, since tests pass well on many runs with or without randomness.